apps/researcher/src/lib/date-formatter/actions.ts

```diff
@@ -0,0 +1,11 @@
+import {getFormatter, getTranslations} from 'next-intl/server';
+import {createFormatter} from './create-formatter';
+
+export async function getFormatDate(): Promise<
+  ReturnType<typeof createFormatter>
+> {
+  const t = await getTranslations('FormatDate');
+  const format = await getFormatter();
```

Comment:
Intuitively I would expect a `getFormatter()` to return a `formatter` (not a `format`). To be sure: what does it return?

Reply:
This is the formatter of [`next-intl`](https://next-intl-docs.vercel.app/docs/usage/dates-times). I followed the examples of `next-intl`, and they call 'it' `format`. This is an object with different formatting functions (`dateTime`, `number`, `relativeTime` and `list`).

I think you are right, we can call this `formatter`.

Reply:
Since we create something similar but with custom format functions, I will rename our functions to `useDateFormatter` and `getDateFormatter`.

I have also added a `useMemo` to `useDateFormatter`, inspired by [next-intl](https://github.com/amannn/next-intl/blob/c6e89c5290c2ce2352926548d16df0bc7b4585e3/packages/use-intl/src/react/useFormatter.tsx).